src/display/canvas.js

```diff
@@ -1359,9 +1359,7 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
       }
 
       var name = fontObj.loadedName || 'sans-serif';
-      var bold = fontObj.black ? (fontObj.bold ? '900' : 'bold') :
-                                 (fontObj.bold ? 'bold' : 'normal');
-
+      var bold = fontObj.black ? '900' : (fontObj.bold ? 'bold' : 'normal');
```

Comment:
Having looked at this once more, ordinary `ArialBlack` should use a font weight that maps to `black` to render correctly (see https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#Common_weight_name_mapping).
However, `ArialBlack,Bold` should ideally map to an *even heavier* font weight, but given that no such one exists (in CSS), we will thus simply fallback to normal `ArialBlack` instead.

To me, this seems to give the best possible rendering (at least on Windows) for these non-embedded fonts; otherwise normal `ArialBlack` actually renders too light.